src/user.rs

```diff
@@ -0,0 +1,46 @@
+use std::fs::File;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+
+use rustc_serialize::json;
+
+use project_types::Project;
+use Error;
+
+
+pub fn new_user<P: AsRef<Path>>(project: String, public: P, name: String) -> Result<(), Error> {
```

Comment:
Impure means it has side effects outside of its parameters and return value.

```
fn add_pure(i32 a, i32 b) -> i32 {
    a + b
}

fn add_impure(i32 a, i32 b) -> i32 {
    us_nukes::launch_missile_at_russia();

    if (us_nukes::is_nuclear_war_happening()) {
        a + b + b // OMG, can't do math during war
    } else {
        a + b
    }
}
```

These both take the same inputs and return the same value, but the first is pure. That means you can call it as many times as you want and it won't be any different from just calling it once. And if you call it with the same parameters 4, 5, you will always get 9.

The second one is impure, because:

1. It doesn't depend only its parameters, it also depends on the implicit state of the world (whether we're in nuclear war or not).
2. It has side effects outside of its return value (it launches a nuke at Russia).

This means if you call it multiple times with the same parameters, its not guaranteed to be the same. It could return different values depending on if we're at nuclear war, and each time you call it, we launch another missile at Russia.

If you get more strict about it, even the first function isn't actually pure, because it has side effects like changing the instruction pointer in the CPU, moves around bits, uses up electricity, etc. But we don't generally consider things like in terms of purity.